NeoML/src/Dnn/Layers/AccuracyLayer.cpp

```diff
@@ -66,10 +67,25 @@ void CAccuracyLayer::RunOnceAfterReset()
 	const int objectCount = inputBlob->GetObjectCount();
 	const int objectSize = inputBlob->GetObjectSize();
 
+	const int expectedDataSize = expectedLabelsBlob->GetDataSize();
+	const int expectedObjectCount = expectedLabelsBlob->GetObjectCount();
+	const int expectedObjectSize = expectedLabelsBlob->GetObjectSize();
+
 	inputBuffer.SetSize( dataSize );
-	expectedBuffer.SetSize( dataSize );
-	inputBlob->CopyTo( inputBuffer.GetPtr(), dataSize );
-	expectedLabelsBlob->CopyTo( expectedBuffer.GetPtr(), dataSize );
+	expectedBuffer.SetSize( expectedDataSize );
+	inputBlob->CopyTo( inputBuffer.GetPtr(), inputBuffer.Size() );
+
+	if( expectedLabelsBlob->GetDataType() == NeoML::CT_Float ) {
+		expectedLabelsBlob->CopyTo( expectedBuffer.GetPtr(), expectedBuffer.Size() );
+	} else {
+		CFastArray<int, 1> expectedBufferInt;
```

Comment:
Use `CDnnBlob::GetBuffer<int>` (and don't forget `CDnnBlob::ReleaseBuffer` afterwards). In that way there will be no data copy (when using CPU).